models/src/node/crd/mod.rs

```diff
@@ -0,0 +1,100 @@
+pub mod v1;
+pub mod v2;
+
+pub use self::v2::{
+    BottlerocketShadow, BottlerocketShadowSpec, BottlerocketShadowState, BottlerocketShadowStatus,
+};
+
+use crate::node::{error, error::Error};
+
+use k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition;
+use kube::CustomResourceExt;
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+use std::fmt::Debug;
+
+lazy_static! {
+    static ref BOTTLEROCKETSHADOW_CRD_METHODS: Vec<fn() -> CustomResourceDefinition> = {
+        // A list of CRD methods for different BottlerocketShadow version
+        // The latest version should be added at the end of the vector.
+        let mut crd_methods = Vec::new();
+        crd_methods.push(v1::BottlerocketShadow::crd as fn()->CustomResourceDefinition);
+        crd_methods.push(v2::BottlerocketShadow::crd as fn()->CustomResourceDefinition);
+        crd_methods
+    };
+}
```

Comment:
I think you could add the result of `combine_crds()` here as a static value, since it should never change.

Reply:
Somehow `serde::ser::Serialize` get lost when I put it in lazy_static, do I would just keep this method as it is.